apps/documents/models.py

```diff
@@ -0,0 +1,26 @@
+from django.db import models
+
+from apps.teams.models import BaseTeamModel
+from apps.utils.conversions import bytes_to_megabytes
+
+
+class RepositoryType(models.TextChoices):
+    COLLECTION = "collection", "Collection"
+
+
+class Repository(BaseTeamModel):
+    name = models.CharField(max_length=255)
+    summary = models.TextField()
+    type = models.CharField(choices=RepositoryType.choices, default=RepositoryType.COLLECTION)
+    files = models.ManyToManyField("files.File", blank=False)
+
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["team", "type", "name"], name="unique_repository_per_team")]
+
+    @property
+    def size(self) -> float:
+        """Returns the size of this repository in megabytes"""
+        return bytes_to_megabytes(sum([bytes or 0 for bytes in self.files.values_list("content_size", flat=True)]))
```

Comment:
Any reason not to do this sum in the database? There's event a [pg_size_pretty](https://www.postgresql.org/docs/11/functions-admin.html#FUNCTIONS-ADMIN-DBOBJECT) function which you could use if you wanted to.

Reply:
No reason, I'll make an update, thanks for the suggestion